sensor/kubernetes/listener/watcher/crd/crdwatcher.go

```diff
@@ -73,44 +69,56 @@ func (w *crdWatcher) AddResourceToWatch(name string) error {
 }
 
 // Watch starts the CRD handler that will dispatch any events coming from k8s related to CRDs to be manage by the CRDWatcher
-func (w *crdWatcher) Watch(statusC chan<- *watcher.Status) error {
-	w.started.Store(true)
+func (w *crdWatcher) Watch() (<-chan *watcher.Status, error) {
+	if w.started.Swap(true) {
+		return nil, errors.New("Watch was already called")
+	}
+	err := w.startHandler()
+	if err != nil {
+		return nil, err
+	}
+	statusC := watch(w.stopSig.Done(), w.resourceC, w.resources.Freeze())
```

Comment:
All arguments are members of `w`, so may be:
```suggestion
	statusC := w.watch()
```